design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,373 @@
+| 1  | *Vanilla flow* - a Conjur identity exists for Azure VM and added to privileged layer         | VM sends authentication request with a valid Azure token | Conjur access token is returned in body and action is logged                                  | Integration               |            |
+| 2  | A Conjur Host identity for Azure VM, added to privileged layer and has permissions on secret | VM sends request with valid Azure token to fetch secret  | Secret is successfully returned and action is logged                                          | Integration               |            |
+| 3  | No Conjur Host identity exists for Azure VM                                                  | VM sends authentication request with a valid Azure token | 403 Forbidden response / RoleNotFound in error body is returned and action is logged          | Integration               |            |
+| 4  | Conjur identity exists for Azure VM but doesn't have permissions on secret                   | VM sends request with valid Azure token to fetch secret  | 401 Unauthorized response is returned and action is logged                                    | Integration               |            |
+
+#### Error handling
+
+|    | Given                                                                                     | When                                                                | Then                                                                                                                                         | Manual / UT / Integration | Status     |
+|----|-------------------------------------------------------------------------------------------|---------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------|---------------------------|------------|
+| 1  | authn-azure is not whitelisted in ENV or enabled in DB                                    | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (AuthenticatorNotWhitelisted) is returned and logged   | Integration               |            |
+| 2  | Policy is loaded w/o "provider-uri" variable or value is missing                          | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (RoleNotAuthorizedOnWebservice) is returned and logged | Integration               |            |
+| 3  | Policy is loaded w/o defining authenticator webservice                                    | VM sends authentication request to Conjur with valid Azure token    | 500 Internal Server Error response with an error body with the relevant error message (WebserviceNotFound) is returned and logged            | Integration               |            |
```

Comment:
1. In this case we return 401
2. We don't return the error message in the body, only in the log